### ISSUE TITLE
undefined: strings.CutPrefix

### ISSUE THREAD

**Issue by Evildethow** [dogmatiq/ferrite#50]
Seems to have broken in either https://github.com/dogmatiq/ferrite/releases/tag/v0.3.3 or https://github.com/dogmatiq/ferrite/releases/tag/v0.3.4

Breaking change is https://github.com/dogmatiq/ferrite/blame/b71f73c276082f04fb0acbbfca19bd5cb0c47d60/internal/markdownmode/ref.go#L33

To reproduce locally  just checked out the project and ran `make`:

```
~/workspace/dogma/ferrite$ make
--- executing core library installation hooks
go test  -race ./...
go: downloading github.com/rivo/uniseg v0.2.0
# github.com/dogmatiq/ferrite/internal/markdownmode
internal/markdownmode/ref.go:33:21: undefined: strings.CutPrefix
FAIL	github.com/dogmatiq/ferrite [build failed]
?   	github.com/dogmatiq/ferrite/internal/limits	[no test files]
FAIL	github.com/dogmatiq/ferrite/internal/markdownmode [build failed]
?   	github.com/dogmatiq/ferrite/internal/validatemode	[no test files]
?   	github.com/dogmatiq/ferrite/maybe	[no test files]
?   	github.com/dogmatiq/ferrite/variable	[no test files]
FAIL
make: *** [.makefiles/pkg/go/v1/Makefile:174: test] Error 2

```

**Comment by Evildethow** [comment]
I tested this using go `1.19.2`:

```
~/workspace/dogma/ferrite$ go version
go version go1.19.2 linux/amd64
```

**Comment by Evildethow** [comment]
Looks like `strings.CutPrefix` was introduced in Go `1.20.x`. I don't see it under `1.19.2` - https://pkg.go.dev/strings@go1.19.2

**Comment by Evildethow** [comment]
FWIW, Ferrite `go.mod` still uses `1.19`, so its not catching the compatiblity issue.

**Comment by jmalloc** [comment]
Cheers, fixed in [v0.3.5](https://github.com/dogmatiq/ferrite/releases/tag/v0.3.5).
